scripts/gen_circleci_config.py

```diff
@@ -21,6 +23,61 @@ def gen_required_suites(template: dict) -> None:
     template["workflows"]["test"]["jobs"].extend([{suite: requires_base_venvs} for suite in required_suites])
 
 
+def gen_pre_checks(template: dict) -> None:
+    """Generate the list of pre-checks that need to be run."""
+    from needs_testrun import pr_matches_patterns
+
+    def check(name: str, command: str, paths: t.Set[str]) -> None:
+        if pr_matches_patterns(paths):
+            template["jobs"]["pre_check"]["steps"].append({"run": {"name": name, "command": command}})
+
+    check(
+        name="Formatting check",
+        command="riot -v run -s fmt && git diff --exit-code",
+        paths={"*.py", "*.pyi"},
```

Comment:
@Yun-Kim an example of how we decide if we should run a check: we provide a set of file patterns. If any of the files within a PR match a pattern then we run the check

Reply:
ahhhh I missed this. Thanks!